Tasks/JavaToolInstaller/FileExtractor/JavaFilesExtractor.ts

```diff
@@ -0,0 +1,149 @@
+        } else if (stats.isDirectory()) {
+            this.failTask(taskLib.loc('ExtractDirFailed', file));
+        }
+
+        if (this.win) {
+            if (".tar" == fileEnding) { // a simple tar
+                    this.sevenZipExtract(file, this.destinationFolder);
+            } else if (".tar.gz" == fileEnding) { // a compressed tar, e.g. 'fullFilePath/test.tar.gz'
+                    // e.g. 'fullFilePath/test.tar.gz' --> 'test.tar.gz'
+                    var shortFileName = file.substring(file.lastIndexOf(path.sep) + 1, file.length);
+                    // e.g. 'destinationFolder/_test.tar.gz_'
+                    var tempFolder = path.normalize(this.destinationFolder + path.sep + '_' + shortFileName + '_');
+                    console.log(taskLib.loc('CreateTempDir', tempFolder, file));
+
+                    // 0 create temp folder
+                    taskLib.mkdirP(tempFolder);
+
+                    // 1 extract compressed tar
+                    this.sevenZipExtract(file, tempFolder);
+                    console.log(taskLib.loc('TempDir', tempFolder));
+                    var tempTar = tempFolder + path.sep + taskLib.ls('-A', [tempFolder])[0]; // should be only one
+                    console.log(taskLib.loc('DecompressedTempTar', file, tempTar));
+                       
+                    // 2 expand extracted tar
+                    this.sevenZipExtract(tempTar, this.destinationFolder);
+
+                    // 3 cleanup temp folder
+                    console.log(taskLib.loc('RemoveTempDir', tempFolder));
+                    taskLib.rmRF(tempFolder);
+            } else if (".zip" == fileEnding) {
+                this.unzipExtract(file, this.destinationFolder);
+            } else { // use sevenZip
+                this.sevenZipExtract(file, this.destinationFolder);
+            }
+        } else { // not windows
+            if (".tar" == fileEnding || ".tar.gz" == fileEnding) {
+                this.tarExtract(file, this.destinationFolder);
+            } else if (".zip" == fileEnding) {
+                this.unzipExtract(file, this.destinationFolder);
+            } else { // fall through and use sevenZip
+                this.sevenZipExtract(file, this.destinationFolder);
+            }
+        }
+    }
+
+    public unzipJavaDownload(repoRoot: string, fileEnding: string) {
+        try {
+            // Create the destination folder if it doesn't exist
+            if (!taskLib.exist(this.destinationFolder)) {
+                console.log(taskLib.loc('CreateDestDir', this.destinationFolder));
+                taskLib.mkdirP(this.destinationFolder);
+            }
+
+            var jdkFile = path.normalize(repoRoot);
+            var stats = taskLib.stats(jdkFile);
+            if (stats.isFile()) {
+                this.extractFiles(jdkFile, fileEnding);
+                taskLib.setResult(taskLib.TaskResult.Succeeded, taskLib.loc('SucceedMsg'));
+            }
+        } catch (e) {
```

Comment:
The task result is set but downstream code will continue executing. Consider just throwiing the error here if we want task execution to be stopped.